### PR TITLE
Feature/qed fvol

### PR DESCRIPTION
**Pull request by jch1g10**
Ported modules MGauge/UnitEm, MScalar/ScalarVP and MScalar/VPCounterTerms to the new Hadrons; combined outputs for each MScalar module into a single output file per configuration; implemented IR improvement and infinite-volume photon.

There are a lot of changes here, but most are from merging with develop and feature/Hadrons. The files I have modified are:
MGauge/{StochEm,UnitEm}.{cc,hpp}
MScalar/{ChargedProp,ScalarVP,VPCounterTerms}.{cc,hpp}
Photon.h

### CODE REVIEW

File: lib/qcd/action/gauge/Photon.h [hunk 1]
```diff
@@ -58,29 +58,34 @@ namespace QCD{
   public:
     INHERIT_GIMPL_TYPES(Gimpl);
     GRID_SERIALIZABLE_ENUM(Gauge, undef, feynman, 1, coulomb, 2, landau, 3);
-    GRID_SERIALIZABLE_ENUM(ZmScheme, undef, qedL, 1, qedTL, 2);
+    GRID_SERIALIZABLE_ENUM(ZmScheme, undef, qedL, 1, qedTL, 2, qedInf, 3);
   public:
-    Photon(Gauge gauge, ZmScheme zmScheme);
+    Photon(Gauge gauge, ZmScheme zmScheme, std::vector<Real> improvements, Real G0);
```
Comment (aportelli) [inline lib/qcd/action/gauge/Photon.h]: Please make overloaded versions with default values for `G0` and `improvements`

**Comment by aportelli** [comment]
Hi James, did you have a chance to look at the suggested changes?

**Comment by jch1g10** [comment]
Hi Antonin. Yes, I've added overloaded constructors for Photon, and merged in the latest changes from feature/hadrons. The Travis build is now failing - seems to be that it doesn't have LIME.

**Comment by aportelli** [comment]
Hi James, there is not much point in keep this `feature/qed-fvol` branch, could you make this request to `feature/hadrons`?

**Comment by aportelli** [comment]
Thanks, I'll wait for the CI to valid and merge. I think from now on you should work from `feature/hadrons`. If you would be keen on spending some time on writing a documentation at https://github.com/aportelli/Hadrons-doc that would be very nice. It does not have to be long.

**Pull request merged**